### PR TITLE
Example emailTask Workflow

### PR DESCRIPTION
**Pull request by rhundal7**
- Sends an email

### CODE REVIEW

File: examples/example-emailtask/nodes/email.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        "spin": true
+      },
+      {
+        "key": "timestamp",
+        "type": "PROCESS"
+      },
+      {
+        "key": "tenantId",
+        "type": "PROCESS"
+      }
```
Comment (kaladay) [inline examples/example-emailtask/nodes/email.json:23]: None of these are being used and should be deleted.
```suggestion
```

Alternatively, we may want to change `mailTo` into a configurable variable and keep the `exampleEmailTo` process variable:

```suggestion
      {
        "key": "exampleEmailTo",
        "type": "PROCESS"
      }
```

File: examples/example-emailtask/nodes/email.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    ],
+    "outputVariable": {},
+    "includeAttachment": false,
+    "mailTo": "{{exampleEmailTo}}",
```
Comment (kaladay) [inline examples/example-emailtask/nodes/email.json:27]:
```suggestion
    "mailTo": "${exampleEmailTo}",
```
If we go with a process variable, then this change is needed.

**Review by kaladay — CHANGES_REQUESTED** [review]

File: examples/example-emailtask/nodes/email.json (hunk 2 above)
Comment (rhundal7) [reply examples/example-emailtask/nodes/email.json]: done

File: examples/example-emailtask/nodes/email.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "mailFrom": "{{exampleEmailFrom}}",
+    "mailMarkup": "<html><head><title>Example Email</title></head><body> <p> Text for Email </p> <body></html>",
```
[the remark this answers is not in the document]
Comment (rhundal7) [reply examples/example-emailtask/nodes/email.json]: done

File: examples/example-emailtask/nodes/email.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      },
+      {
+        "key": "tenantId",
+        "type": "PROCESS"
+      }
+    ],
+    "outputVariable": {},
+    "includeAttachment": false,
+    "mailTo": "{{exampleEmailTo}}",
+    "mailFrom": "{{exampleEmailFrom}}",
```
[the remark this answers is not in the document]
Comment (rhundal7) [reply examples/example-emailtask/nodes/email.json]: Initial commit pushed - wip

File: examples/example-emailtask/nodes/email.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      "type": "PROCESS"
+    },
+    {
+      "key": "timestamp",
+      "type": "PROCESS"
+    },
+    {
+      "key": "tenantId",
+      "type": "PROCESS"
+    }
```
Comment (kaladay) [inline examples/example-emailtask/nodes/email.json:18]:
```suggestion
    }
```
I don't think the `timestamp` and `tenantId` are needed.

File: examples/README.md [hunk 6]
```diff
@@ -17,3 +17,63 @@ The `wd` variable in the `fw-cli` configuration should then look something like
...
+fw config set exampleEmailTo ***
+```
+
+These variables are required when triggering the workflow:
+
+| Variable Name  | Allowed Values | Short Description |
+| -------------- | -------------- | ----------------- |
+| emailTo        | e-mail address | An e-mail address used as the "TO" in the sent e-mails. |
+| username       | string         | Okapi login username. |
+| password       | string         | Okapi login password. |
```
Comment (kaladay) [inline examples/README.md:56]:
```suggestion
| username       | string         | Okapi login username. |
| password       | string         | Okapi login password. |
```

I don't think these are needed for sending an e-mail.

File: examples/README.md [hunk 7]
```diff
@@ -17,3 +17,63 @@ The `wd` variable in the `fw-cli` configuration should then look something like
...
+
+## example-emailtask
+
+### Example EmailTask Workflow
+
+This workflows sends an email to the user who's email address is specified in the config file or as a user input. 
+
+```shell
+fw config set exampleEmailFrom ***
+fw config set exampleEmailTo ***
```
Comment (kaladay) [inline examples/README.md:47]:
```suggestion
```
This is no longer a fw config parameter and is a request parameter.
Please remove.

File: examples/README.md [hunk 8]
```diff
@@ -17,3 +17,63 @@ The `wd` variable in the `fw-cli` configuration should then look something like
...
+```
+
+These variables are required when triggering the workflow:
+
+| Variable Name  | Allowed Values | Short Description |
+| -------------- | -------------- | ----------------- |
+| emailTo        | e-mail address | An e-mail address used as the "TO" in the sent e-mails. |
```
Comment (kaladay) [inline examples/README.md:54]:
```suggestion
| Variable Name    | Allowed Values | Short Description |
| ---------------- | -------------- | ----------------- |
| exampleEmailFrom | e-mail address | The source e-mail address to send from. |
| exampleEmailTo   | e-mail address | The destination e-mail address to send to. |
```

The table should describe the properties the task is needing.

**Review by kaladay — CHANGES_REQUESTED** [review]

File: examples/README.md [hunk 9]
```diff
@@ -17,3 +17,61 @@ The `wd` variable in the `fw-cli` configuration should then look something like
...
+fw config set exampleEmailFrom ***
+```
+
+These variables are required when triggering the workflow:
+
+| Variable Name  | Allowed Values | Short Description |
+| -------------- | -------------- | ----------------- |
+| exampleEmailFrom | e-mail address | The source e-mail address to send from. |
+| exampleEmailTo   | e-mail address | The destination e-mail address to send to. |
+| logLevel       | [INFO,DEBUG]   | Desired log level |
```
Comment (kaladay) [inline examples/README.md:55]:
```suggestion
| Variable Name    | Allowed Values | Short Description |
| ---------------- | -------------- | ----------------- |
| exampleEmailFrom | e-mail address | The source e-mail address to send from. |
| exampleEmailTo   | e-mail address | The destination e-mail address to send to. |
| logLevel         | [INFO,DEBUG]   | Desired log level |
```

File: examples/README.md [hunk 10]
```diff
@@ -17,3 +17,61 @@ The `wd` variable in the `fw-cli` configuration should then look something like
...
+fw run example-emailtask
+```
+
+Trigger the workflow using an **HTTP** request such as with **Curl**:
+
+```shell
+curl --location --request POST 'https://folio-edge-dev.library.tamu.edu/mod-workflow/events/workflow/example-emailtask/start' \
+  --header 'Content-Type: application/json' \
+  --header 'X-Okapi-Tenant: diku' \
+  --data-raw '{ "logLevel": "INFO", "exampleEmailFrom": "*", "username":"diku","password":"*", "exampleEmailTo": "you@example.com" }'
```
Comment (kaladay) [inline examples/README.md:75]: Don't put the TAMU specific URL here.
It should be something like `http://localhost:9001/` ...

Also, the example data needs to be reasonably correct.
The configuration is wrong.
There should be no `username`, `password`, or `exampleEmailFrom`.

Try something like this:

```suggestion
curl --location --request POST 'http://localhost:9001/mod-workflow/events/workflow/example-emailtask/start' \
  --header 'Content-Type: application/json' \
  --header 'X-Okapi-Tenant: diku' \
  --data-raw '{ "logLevel": "INFO", "exampleEmailTo": "you@example.com" }'
```

**Review by kaladay — CHANGES_REQUESTED** [review]

**Review by kaladay — APPROVED** [review]

**Pull request merged**